src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
In this case, you have two tasks
1) find children
2) calculate the sum
More informative example ->
`
children.forEach((ch) => {
 const SOME_VARIABLE_NAME = people.find((m) => m.name === ch.mother).born
 sum += ch.born - SOME_VARIABLE_NAME
});
`
But you cannot use forEach method in this task, you need to rewrite, using the method reduce. As described in the comment above